blockchain/blockdao.go

```diff
@@ -606,6 +701,129 @@ func (dao *blockDAO) deleteTipBlock() error {
 	return dao.kvstore.Commit(batch)
 }
 
+// getDBFromHash returns db of this block stored
+func (dao *blockDAO) getDBFromHash(h hash.Hash256) (db.KVStore, uint64, error) {
+	hei, err := dao.getBlockHeight(h)
+	if err != nil {
+		return nil, 0, err
+	}
+	return dao.getDBFromHeight(hei, blockHeightToFileKey)
+}
+
+func (dao *blockDAO) getTopDB(blkHeight uint64) (kvstore db.KVStore, index uint64, err error) {
+	if dao.cfg.SplitDBSize == 0 {
+		return dao.kvstore, 0, nil
+	}
+	if blkHeight <= dao.cfg.SplitDBHeight {
+		return dao.kvstore, 0, nil
+	}
+	topIndex := dao.topIndex.Load().(uint64)
+	file, dir := getFileNameAndDir(dao.cfg.DbPath)
+	if err != nil {
+		return
+	}
+	longFileName := dir + "/" + file + fmt.Sprintf("-%08d", topIndex) + ".db"
+	dat, err := os.Stat(longFileName)
```

Comment:
Can you measure the overhead of this call, which seems to have to execute on every put block now